src/vos/vos_obj.c

```diff
@@ -247,6 +247,16 @@ key_iter_fetch(struct vos_obj_iter *oiter, vos_iter_entry_t *ent,
 		else
 			ent->ie_epoch = DAOS_EPOCH_MAX;
 		ent->ie_earliest = rbund.rb_krec->kr_earliest;
+		if (oiter->it_iter.it_type == VOS_ITER_AKEY) {
+			if (rbund.rb_krec->kr_bmap & KREC_BF_EVT) {
+				ent->ie_child_type = VOS_ITER_RECX;
+			} else {
+				D_ASSERT(rbund.rb_krec->kr_bmap & KREC_BF_BTR);
+				ent->ie_child_type = VOS_ITER_SINGLE;
+			}
+		} else {
+			ent->ie_child_type = VOS_ITER_AKEY;
+		}
```

Comment:
So we don't support coexisted SV and Array for an akey anymore? That's not what I got from the Jira ticket description.
There are few tests cases for coexisted SV and Array in the evtree aggregation patch (231)

Reply:
Yeah, you're right.   I was somewhat lazy in using an old ticket for the unrelated problem of not exporting information about the subtrees.   But we did talk about making this change as we couldn't come up with a use case for supporting it and it's somewhat wasteful (though maybe not the lowest hanging fruit in that respect.

The one caveat is that when we move to logging punch, you would also either need to log this or have typed keys that can't be untyped until after a removal.   But Johann was OK with that limitation as we don't really have any use case for mixed types.

Reply:
Though I guess if we are logging creates and punches, no reason we couldn't attach the flags

Reply:
I see, thanks for the background information.